### PR TITLE
[Sharing] Closing feature 8728 : no more only for admins of apps

### PR DESCRIPTION
**Pull request by NicolasEYSSERIC**
Removing obsolete behavior and property : 'ui.useContextualMenu' of Attachment.properties
Columns of ArrayPane made by tags can be sort (adding new attribute 'compareOn')
Adding new column 'Creation date' in list of user shares. By default, this list is now sorted accorded to this date.

Do not forget to check out PR on Silverpeas-Components...

### CODE REVIEW

File: core-library/src/main/java/org/silverpeas/core/contribution/publication/model/PublicationDetail.java [hunk 1]
```diff
@@ -62,8 +63,10 @@
 import org.silverpeas.core.security.authorization.AccessController;
 import org.silverpeas.core.security.authorization.AccessControllerProvider;
 import org.silverpeas.core.security.authorization.PublicationAccessControl;
+import org.silverpeas.core.security.authorization.SimpleDocumentAccessControl;
```
Comment (silveruser) [inline core-library/src/main/java/org/silverpeas/core/contribution/publication/model/PublicationDetail.java]: MINOR Remove this unused import 'org.silverpeas.core.security.authorization.SimpleDocumentAccessControl'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

File: core-web/src/main/java/org/silverpeas/core/web/attachment/tag/SimpleDocumentContextualMenu.java [hunk 2]
```diff
@@ -220,29 +197,84 @@ String prepareActions(SimpleDocument attachment, boolean useXMLForm, boolean use
...
+
+    String itemsConfig = builder.toString();
+
+    return getMenu(attachmentId, menuItems, itemsConfig);
+  }
+
+  String prepareReadOnlyActions(SimpleDocument attachment, UserDetail user,
+      LocalizationBundle resources, boolean showMenuNotif) throws UnsupportedEncodingException {
+    String attachmentId = String.valueOf(attachment.getOldSilverpeasId());
+    StringBuilder itemsBuilder = new StringBuilder(2048);
```
Comment (silveruser) [inline core-web/src/main/java/org/silverpeas/core/web/attachment/tag/SimpleDocumentContextualMenu.java]: MINOR Assign this magic number 2048 to a well-named constant, and use the constant instead. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS109)

File: core-web/src/main/java/org/silverpeas/core/web/attachment/tag/SimpleDocumentContextualMenu.java [hunk 3]
```diff
@@ -220,29 +197,84 @@ String prepareActions(SimpleDocument attachment, boolean useXMLForm, boolean use
...
+    configBuilder.append(String.format(TEMPLATE, attachmentId, "0", !showMenuNotif));
+    if (sharingAllowed) {
+      configBuilder.append(String.format(TEMPLATE, attachmentId, "1", false));
     }
 
+    return getMenu(attachmentId, itemsBuilder.toString(), configBuilder.toString());
+  }
+
+  private String getMenu(String attachmentId, String items, String config) {
+    StringBuilder builder = new StringBuilder(2048);
```
Comment (silveruser) [inline core-web/src/main/java/org/silverpeas/core/web/attachment/tag/SimpleDocumentContextualMenu.java]: MINOR Assign this magic number 2048 to a well-named constant, and use the constant instead. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS109)

**Comment by silveruser** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by SilverYoCha** [comment]
Merged.
Closed manually because of manual git operations performed during integration.

**Pull request closed without merging**